services/slam/builtin/cartographer_int_test.go

```diff
@@ -0,0 +1,150 @@
+package builtin_test
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"testing"
+
+	"github.com/edaniels/golog"
+	"go.viam.com/rdk/services/slam"
+	"go.viam.com/rdk/services/slam/builtin"
+	"go.viam.com/rdk/services/slam/internal"
+	"go.viam.com/test"
+	"go.viam.com/utils"
+	"go.viam.com/utils/artifact"
+)
+
+// Creates the lua files required by the cartographer binary.
+func createLuaFiles(name string) error {
+	if err := os.Mkdir(name+"/config/lua_files", os.ModePerm); err != nil {
+		return err
+	}
+	for _, luaFile := range []string{"locating_in_map.lua", "mapping_new_map.lua", "updating_a_map.lua"} {
+		source, err := os.Open(artifact.MustPath("slam/" + luaFile))
+		if err != nil {
+			return err
+		}
+		defer source.Close()
+		destination, err := os.Create(name + "/config/lua_files/" + luaFile)
+		if err != nil {
+			return err
+		}
+		defer destination.Close()
+		_, err = io.Copy(destination, source)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// Checks the cartographer position and map.
+func testCartographerPositionAndMap(t *testing.T, svc slam.Service) {
+	t.Helper()
+
+	actualMIME, _, pointcloud, err := svc.GetMap(context.Background(), "test", "pointcloud/pcd", nil, false)
+	test.That(t, err, test.ShouldBeNil)
+	test.That(t, actualMIME, test.ShouldResemble, "pointcloud/pcd")
+	test.That(t, pointcloud.Size(), test.ShouldEqual, 200)
+
```

Comment:
[q] Why are we not testing here for the retrieval of the JPEG GetMap? If we did, wouldn't it also be an issue that it would take a long time to retrieve it?

Reply:
For orbslam, I only tested retrieval of the pointcloud. I'm not sure what we could test about the jpeg map, other than that we get a response. I think it's okay to rely on manual testing for the jpeg map. I'm not sure if it would be an issue that it would take a long time to retrieve it--I don't know if our networking code has any default request timeouts.